### DISCUSSION TITLE
Failed to router for MPA with history mode

### DISCUSSION THREAD

**Discussion by cn-ryh** [vuejs/router#2217] [Help and Questions] [answered]
I am developing a multi-page application that uses vue-router4 history mode for routing. When I use `npm run serve`, the access is normal. However, when I use `live- server` or `http- server` to access the builded `dist` folder, for pages that do not have secondary routing (for example, access to `localhost:8081/login` can be accessed normally), but for those pages with secondary routing (such as `localhost:8081/user/1`), It shows CANNOT FOUND /user/1 occurs. 
Obviously this is because there is no access to the routing program at all. 
It is worth mentioning that when I use hash (for example, `localhost:8081/user/#/1`), multi-level routing can be accessed normally.

Here is my code of router in `user` dictory.

```ts
import * as VueRouter from 'vue-router';
import training from './user-detail.vue';

const routes = [
    { path: '/:id', component: training },
];

const router = VueRouter.createRouter({
    history: VueRouter.createWebHistory(`user`),
    routes,
});

export { router };
```

**Comment by posva** [comment, marked as answer]
You need to configure the server to still return the index.html page: https://router.vuejs.org/guide/essentials/history-mode.html#Example-Server-Configurations
